### PR TITLE
Use TF backend docstrings for theano docstrings if theano undocumented

### PR DESCRIPTION
**Pull request by bstriner**
The theano backend currently has 69 functions that are undocumented but documented in the TF backend. However, we don't want to replicate a lot of documentation. This PR doesn't affect Keras on TF. However, on Theano backend, this PR fills in missing docstrings from TF. If the function is documented in theano_backend, it is not affected, but if it is undocumented in theano_backend and documented in tensorflow_backend, the documentation is copied over.

I changed `tensorflow_backend.py` to conditionally import tensorflow.
* If you set keras to tensorflow and do not have tensorflow installed, importing tensorflow_backend will fail with an import error as expected.
* If you set keras to theano and do not have tensorflow installed, you can still import tensorflow_backend. You can't run any of the functions but can still access dostrings.

I updated backend `__init__.py` to have a for loop that temporarily imports tensorflow_backend, and for any function documented there but not in theano_backend, copies the docstring, then deletes the reference to the tensorflow backend. 

The end result is that something like `help(K.binary_crossentropy)` will show documentation in both theano and tensorflow. Additionally, running autogen and mkdocs will show meaningful documentation with both backends.

As a result of this PR, you could delete any docstrings in theano_backend that are not intentionally meant to indicate a difference. The only docstrings in theano_backend should be those where someone wants to specifically describe a difference between the backends.

Cheers!

### CODE REVIEW

**Comment by bstriner** [comment]
These are the current list of 69 functions that are documented in tensorflow but not in theano, that have their documentation filled in by this PR. There are also several functions in theano backend that look like they have old documentation that could be deleted so it will be replaced by the current tensorflow documentation.

Also, in view of https://github.com/fchollet/keras/pull/5118, this means autogen.py --validate could be used to check either backend.

- l2_normalize
- sigmoid
- random_uniform_variable
- binary_crossentropy
- random_binomial
- pow
- set_value
- greater
- tile
- lesser_equal
- softmax
- round
- square
- argmax
- softsign
- sign
- flatten
- set_learning_phase
- batch_set_value
- sparse_categorical_crossentropy
- abs
- random_uniform
- concatenate
- transpose
- learning_phase
- equal
- hard_sigmoid
- softplus
- ones_like
- ndim
- in_test_phase
- pool2d
- dtype
- atrous_conv2d
- log
- separable_conv2d
- relu
- var
- function
- zeros_like
- cos
- in_train_phase
- random_normal_variable
- std
- not_equal
- argmin
- tanh
- to_dense
- maximum
- lesser
- cast
- minimum
- random_normal
- clip
- pool3d
- gradients
- min
- get_value
- sqrt
- sin
- max
- get_variable_shape
- stack
- is_sparse
- greater_equal
- exp
- reshape
- categorical_crossentropy
- dot

**Comment by bstriner** [comment]
BTW, I moved the definition of backend() up so it could be used from the tensorflow backend.

**Comment by bstriner** [comment]
Slight fix to the logic. `tensorflow_backend.py` always tries to import tensorflow. If there is an import error and tensorflow is the backend, an import exception is raised. If there is an import error but theano is the backend, the exception is eaten and exceptions will only be raised if you try to then run a function that depends on tensorflow.

- `import tensorflow_backend` is unchanged if tensorflow is the backend or if tensorflow is installed
- `import tensorflow_backend` now works if theano is the backend and tensorflow is uninstalled, it will just throw an exception when you try to run a function that depends on the module

**Comment by fchollet** [comment]
Closing outdated PR. If you still care about the content of the PR, please submit a new PR to `master`, updated for the Keras 2.0 API.
